src/components/app/TabBar.js

```diff
@@ -24,14 +24,14 @@ type Props = {
   onChangeTabOrder: (number[]) => Action,
 };
 
-class TabBar extends PureComponent {
+class TabBar extends PureComponent<Props> {
   constructor(props: Props) {
     super(props);
     (this: any)._mouseDownListener = this._mouseDownListener.bind(this);
   }
 
-  _mouseDownListener(e: SyntheticMouseEvent & { target: HTMLElement }) {
-    this.props.onSelectTab(e.target.dataset.name);
+  _mouseDownListener(e: SyntheticMouseEvent<HTMLElement>) {
```

Comment:
This is the new way of handling events where we want to specially type the `currentTarget`. Note also that we should use `currentTarget` instead of `target`.
All of this is well explained in https://flow.org/en/docs/react/events/

Reply:
Great, looks like a nice update.